File: python/isbn-verifier/isbn_verifier.py
import re

def is_valid(isbn):
    normalized = isbn.replace("-", "")
    if len(normalized) != 10 or not (normalized.isdigit() or normalized[-1] == "X"):
        return False
        
    values = []
    for index, char in enumerate(normalized):
        if char.isdigit():
            values.append(int(char) * (10 - index))
        else:
            values.append(10)
            
    return sum(values) % 11 == 0

File: clojure/isbn-verifier/src/isbn_verifier.clj
(ns isbn-verifier
  (:require [clojure.string :as str]))

(defn place-value
  [place isbn-char]
  (if (= \X isbn-char)
    10
    (* (-> isbn-char str parse-long) (- 10 place))))

(defn isbn?
  [isbn]
  (let [normalized (str/replace isbn #"-" "")
        check-char (last normalized)]
    (and (= 10 (count normalized))
         (every? #{\1 \2 \3 \4 \5 \6 \7 \8 \9 \0 \X} normalized)
         (or (Character/isDigit check-char) (= \X check-char))
         (-> normalized
             (->> (map-indexed place-value)
                  (reduce +))
             (mod 11)
             zero?))))
